app/routes.py

```diff
@@ -1,2 +1,345 @@
-from flask import Blueprint
+from app import db
+from flask import request, Blueprint, jsonify
+from .models.task import Task
+from .models.goal import Goal
+import datetime
+import requests
+import os
+
+
+### WAVES 1 AND 2 - CRUD for TASKS
+
+# Create an endpoint specifically for tasks (all the routes start with /tasks):
+tasks_bp = Blueprint("tasks", __name__, url_prefix="/tasks")
+
+def is_int(value):
+    try:
+        return int(value)
+    except ValueError:
+        return False
+
+
+# define a route with default empty string to GET all tasks:
+@tasks_bp.route("", methods=["GET"], strict_slashes=False)
+def get_tasks():
+    sort_query = request.args.get("sort")
+    if sort_query == "asc":
+        tasks = Task.query.order_by("title")
+    elif sort_query == "desc":
+        tasks = Task.query.order_by(Task.title.desc())  
+    else:
+        tasks = Task.query.all()
+
+    tasks_response = []
+
+    for task in tasks:
+        if task.completed_at == None:
+            tasks_response.append({
+            "id": task.id,
+            "title": task.title,
+            "description": task.description,
+            "is_complete": False
+        })
+        else:
+            tasks_response.append({
+            "id": task.id,
+            "title": task.title,
+            "description": task.description,
+            "is_complete": True
+            })
+
+    return jsonify(tasks_response), 200
+
+
+# define a route with default empty string for POST:
+@tasks_bp.route("", methods=["POST"], strict_slashes=False)
+def create_task():  
+
+    request_body = request.get_json()
+
+    if not request_body.get("title") or not request_body.get("description"):
+        return jsonify({"details": "Invalid data"}), 400 
+
+    try:
+        request_body["completed_at"] == None
+    except KeyError:
+        return jsonify({"details": "Invalid data"}), 400 
+
+    new_task = Task(title = request_body["title"], \
+        description = request_body["description"], \
+        completed_at = request_body["completed_at"])
+
+    db.session.add(new_task)   
+    db.session.commit()  
+
+    # we need an extra step to transfer completed_at to is_completed:
+    task_dict = new_task.to_dict()
+
+    return jsonify({"task": task_dict}), 201
+
+
+# define a new route to GET a specific task
+@tasks_bp.route("/<task_id>", methods=["GET"], strict_slashes=False)
+def get_one_task(task_id):
+
+    if not is_int(task_id):        
+        return {"message": f"ID {task_id} must be an integer"}, 400
+    
+    task = Task.query.get(task_id)
+
+    if task is None:
+        return jsonify(None), 404
+    else:
+        task_dict = task.to_dict()
```

Comment:
Flask has a method called `.get_or_404()` that can reduce the following lines 

```suggestion
    task = Task.query.get_or_404(task_id)
    task_dict = task.to_dict()
```